### ISSUE TITLE
hostkey_checking is global 

### ISSUE THREAD

**Issue by yhenriet** [bluebanquise/bluebanquise#537]
Hello, 
It seems that the hostkey settings is set globaly for every host
from inventory/group_vars/all/general_settings/security.yml:
```
---
security:
  ssh:
    hostkey_checking: true # enable or disable hostkey checking
```

We can add the var to the host , but the role ssh_master is not using it in this way :

From roles/core/ssh_master/templates/config.j2 : 

```
Host {{ host }}
    {% if **security.ssh.hostkey**_checking is not defined or
          security.ssh.hostkey_checking == false %}
    StrictHostKeyChecking no
    UserKnownHostsFile=/dev/null
    {% endif %}
```
So we can't only define the host check for only one host for example . 

But may be we are thinking wrong :)

**Comment by btravouillon** [comment]
Hi @yhenriet ,

Would the change at https://github.com/bluebanquise/bluebanquise/compare/master...btravouillon:fix/537 address your need?

**Comment by oxedions** [comment]
It should work:

```
{% macro write_host(host,macro_node_main_network,ssh_jump) %}
Host {{ host }}
    {% if hostvars[host].ssh_master_stricthostkeychecking is defined %}
    StrictHostKeyChecking {{ hostvars[host].ssh_master_stricthostkeychecking }}
    {% endif %}
    {% if hostvars[host].ssh_master_userknownhostsfile is defined %}
    UserKnownHostsFile {{ hostvars[host].ssh_master_userknownhostsfile }}
```

Will apply to hosts that you desire to exclude from the global configuration, while the last part:

```
Host *
{% if security.ssh.hostkey_checking is not defined or
      security.ssh.hostkey_checking == false %}
    StrictHostKeyChecking no
    UserKnownHostsFile /dev/null
{% endif %}
```

only apply to all hosts that do not have these parameters set. I just do not know if ssh when reading this file will precedence `Host myhost` parameters over `Host *`. I assume @btravouillon already tested and validated it.

**Comment by btravouillon** [comment]
> I"m trying to avoid the message in clush for the computes nodes . So to not have the checking with compute , but to keep it for others .

What message are you trying to avoid?
Note that for production system, `StrictHostKeyChecking` should never be set to `no`.

`security.ssh.hostkey_checking` is indeed a parameter of the target host where the role `ssh_master` run, to enable or disable host key checking for all hosts configured in `~/.ssh/config` of the target host. We won't change the behavior, this would introduce a breaking change.

My proposal is to configure this as global defaults for all hosts by moving it to `Host *` (see ssh_config(5)).

Then, you can define custom `StrictHostkeyChecking`/`UserKnownHostsFile` for each host with `ssh_master_stricthostkeychecking` and `ssh_master_userknownhostsfile` in your inventory.

Example: configure `StrictHostKeyChecking accept-new` for compute nodes only, default (ask) for others.

```
mgmt# cp resources/examples/simple_cluster/inventory/group_vars/all/general_settings/security.yml \
 /etc/bluebanquise/inventory/group_vars/all/general_settings/
```
```
mgmt# cat > /etc/bluebanquise/inventory/group_vars/mg_computes/ssh_master.yml <<EOF
ssh_master_stricthostkeychecking: accept-new
EOF
```
```
mgmt# ansible-playbook playbooks/managements.yml --limit management1 --tags ssh_master
```
```
mgmt# cat /root/.ssh/config


#### Blue Banquise file ####
## Ansible managed file, do not edit


Host *

Host c001
    StrictHostKeyChecking accept-new
    Hostname %h-ice1-1

Host c002
    StrictHostKeyChecking accept-new
    Hostname %h-ice1-1

Host c003
    StrictHostKeyChecking accept-new
    Hostname %h-ice1-1

Host c004
    StrictHostKeyChecking accept-new
    Hostname %h-ice1-1

Host login1
    Hostname %h-ice1-1

Host management1
    Hostname %h-ice1-1

Host management2
    Hostname %h-ice1-1
```

**Comment by btravouillon** [comment]
Hello @yhenriet ,

Were you able to test the code proposal?
Should I open a PR with this change?

**Comment by MarbolanGos** [comment]
This is like the comment I have done here: https://github.com/bluebanquise/bluebanquise/issues/557

I will see if I can patch what @btravouillon offered.

**Comment by btravouillon** [comment]
Please let me know if it is of interest and I will open a PR.

**Comment by oxedions** [comment]
Fixed in PR #638
